src/config/schema.ts

```diff
@@ -11,73 +15,104 @@ export const configSchema = z.object({
+      debug: defaultBoolean,
     })
-    .default({}),
-  postgres: z.object({
-    // connection URL for postgres database
-    connection: z.string(),
+    .default({
+      format: 'pretty',
+      debug: 'false',
+    }),
+  postgres: z
+    .object({
+      // connection URL for postgres database
+      connection: z.string(),
 
-    // run all migrations on boot of the application
-    migrateOnBoot: z.coerce.boolean().default(false),
+      // run all migrations on boot of the application
+      migrateOnBoot: defaultBoolean,
 
-    // try to sync the schema on boot, useful for development
-    // will always keep the database schema in sync with the connected database
-    // it is extremely destructive, do not use it EVER in production
-    syncSchema: z.coerce.boolean().default(false),
+      // try to sync the schema on boot, useful for development
+      // will always keep the database schema in sync with the connected database
+      // it is extremely destructive, do not use it EVER in production
+      syncSchema: defaultBoolean,
 
-    // Enable debug logging for MikroORM - Outputs queries and entity management logs
-    // Do NOT use in production, leaks all sensitive data
-    debugLogging: z.coerce.boolean().default(false),
+      // Enable debug logging for MikroORM - Outputs queries and entity management logs
+      // Do NOT use in production, leaks all sensitive data
+      debugLogging: defaultBoolean,
 
-    // Enable SSL for the postgres connection
-    ssl: z.coerce.boolean().default(false),
-  }),
-  crypto: z.object({
-    // session secret. used for signing session tokens
-    sessionSecret: z.string().min(32),
-  }),
-  meta: z.object({
-    // name and description of this backend
-    // this is displayed to the client when making an account
-    name: z.string().min(1),
-    description: z.string().min(1).optional(),
-  }),
+      // Enable SSL for the postgres connection
+      ssl: defaultBoolean,
+    })
+    .default({
+      connection: '',
+      migrateOnBoot: 'false',
+      syncSchema: 'false',
+      debugLogging: 'false',
+      ssl: 'false',
+    }),
+  crypto: z
+    .object({
+      // session secret. used for signing session tokens
+      sessionSecret: z.string().min(32),
+    })
+    .default({
+      sessionSecret: 'ThisIsADefaultSecretSessionKeyPlaceholder',
+    }),
+  meta: z
+    .object({
+      // name and description of this backend
+      // this is displayed to the client when making an account
+      name: z.string().min(1),
+      description: z.string().min(1).optional(),
+    })
+    .default({
+      name: ' ',
+    }),
   captcha: z
     .object({
       // enabled captchas on register
-      enabled: z.coerce.boolean().default(false),
+      enabled: defaultBoolean,
 
       // captcha secret
       secret: z.string().min(1).optional(),
 
       clientKey: z.string().min(1).optional(),
     })
-    .default({}),
+    .default({
+      enabled: 'false',
+    }),
```

Comment:
Instead of doing this for everything, we can do `defaultBoolean.default("false")` and maybe rename `defaultBoolean` to `booleanSchema`

```suggestion
    .default({
      enabled: 'false',
    }),
```